core/kernel/Integer_spec.rb

```diff
@@ -99,6 +99,51 @@
   it "raises a FloatDomainError when passed Infinity" do
     lambda { Integer(infinity_value) }.should raise_error(FloatDomainError)
   end
+
+  ruby_version_is "2.6" do
+    describe "when passed exception: false" do
+      describe "and to_i returns a value that is not an Integer" do
+        it "swallows an error" do
+          obj = mock("object")
+          obj.should_receive(:to_i).and_return("1")
+          Integer(obj, exception: false).should == nil
+        end
+      end
+
+      describe "and no to_int or to_i methods exist" do
+        it "swallows an error" do
+          obj = mock("object")
+          Integer(obj, exception: false).should == nil
+        end
+      end
+
+      describe "and to_int returns nil and no to_i exists" do
+        it "swallows an error" do
+          obj = mock("object")
+          obj.should_receive(:to_i).and_return(nil)
+          Integer(obj, exception: false).should == nil
+        end
+      end
+
+      describe "and passed NaN" do
+        it "raises a TypeError" do
+          lambda { Integer(nan_value, exception: false) }.should raise_error(FloatDomainError)
+        end
+      end
+
+      describe "and passed Infinity" do
+        it "raises a TypeError" do
+          lambda { Integer(infinity_value, exception: false) }.should raise_error(FloatDomainError)
+        end
+      end
+
+      describe "and passed nil" do
+        it "swallows an error" do
+          Integer(nil, exception: false).should == nil
+        end
+      end
```

Comment:
Could add a spec where it's a valid input, like `"123"` and one where it's an invalid String, like `"abc"`?